Slider/Assets/Scripts/Map/Jungle/Path.cs

```diff
@@ -225,42 +226,59 @@ public void ChangePair()
 
         Physics2D.queriesStartInColliders = false;
 
-        RaycastHit2D checkOne = Physics2D.Raycast(transform.position, one.normalized, 7, LayerMask.GetMask("JunglePaths"));
-        RaycastHit2D checkTwo = Physics2D.Raycast(transform.position, two.normalized, 7, LayerMask.GetMask("JunglePaths"));
+        RaycastHit2D[] checkOne = Physics2D.RaycastAll(transform.position, one.normalized, (1 + length / 2), LayerMask.GetMask("JunglePaths"));
+        RaycastHit2D[] checkTwo = Physics2D.RaycastAll(transform.position, two.normalized, (1 + length / 2), LayerMask.GetMask("JunglePaths"));
 
-        //want to find the closest bin or box and stile
-        if (checkOne.collider != null)
+        foreach (RaycastHit2D hit in checkOne)
         {
-            Path other = checkOne.collider.gameObject.GetComponent<Path>();
-            if (!other.transform.parent.Equals(this.transform.parent))
+            if (hit.collider != null)
             {
-                if (this.gameObject.name == "house right")
-                {
-                    // print(other.gameObject.name);
-                }
-                other.pair = this;
-                pair = other;
-
-                if (this.gameObject.name == "house right")
+                Path other = hit.collider.gameObject.GetComponent<Path>();
+                if (!other.transform.parent.Equals(this.transform.parent) && this.horizontal == other.horizontal)
                 {
-                    // print(other.pair);
+                    other.pair = this;
+                    pair = other;
+                    break;
                 }
             }
         }
-        if (checkTwo.collider != null && pair == null)
+        if (pair == null)
         {
-            Path other = checkTwo.collider.gameObject.GetComponent<Path>();
-            if (!other.transform.parent.Equals(this.transform.parent))
+            foreach (RaycastHit2D hit in checkTwo)
             {
-                if (this.gameObject.name == "house right")
+                if (hit.collider != null)
                 {
-                    print(other.gameObject.name);
+                    Path other = hit.collider.gameObject.GetComponent<Path>();
+                    if (!other.transform.parent.Equals(this.transform.parent))
+                    {
+                        other.pair = this;
+                        pair = other;
+                    }
                 }
-                other.pair = this;
-                pair = other;
             }
         }
 
         Physics2D.queriesStartInColliders = true;
     }
+
+    void OnDrawGizmosSelected()
```

Comment:
can we make this `onDrawGizmos` instead? or even better extract this to a method and add a bool we can use to toggle it on/off from the always draw method (and make it always draw on selected)